Handle OpenSSL 3.2.0 new error message for unexpected unencrypted records

### Subject

I was expecting to all tests working on "main" branch but there are 3 failing test cases

* TestProxyManager::test_https_proxymanager_connected_to_http_proxy[http]
* TestProxyManager::test_https_proxymanager_connected_to_http_proxy[https]
* TestSSL::test_ssl_failure_midway_through_conn 

```
nox -s test-3.12 
...
FAILED test/with_dummyserver/test_socketlevel.py::TestProxyManager::test_https_proxymanager_connected_to_http_proxy[http] - assert 'Your proxy appears to only use HTTP and not HTTPS' in "('Unable to connect to proxy', SSLError(SSLError(1, '[SSL] record layer failure (_ssl.c:1000)')))"
FAILED test/with_dummyserver/test_socketlevel.py::TestProxyManager::test_https_proxymanager_connected_to_http_proxy[https] - assert 'Your proxy appears to only use HTTP and not HTTPS' in "('Unable to connect to proxy', SSLError(SSLError(1, '[SSL] record layer failure (_ssl.c:1000)')))"
FAILED test/with_dummyserver/test_socketlevel.py::TestSSL::test_ssl_failure_midway_through_conn - AssertionError: Regex pattern did not match.
```

### Environment

```
OS macOS-14.2.1-x86_64-i386-64bit
Python 3.12.1
OpenSSL 3.2.0 23 Nov 2023
urllib3 2.1.0
```
### Steps to Reproduce

```
nox -s test-3.12 
```

### Expected Behavior

The test should work, or they should be skipped, or if it's because they are not supported on macOS, then may it should be conditionally skipped. 


### Actual Behavior

3 test fails

--- a/changelog/3268.bugfix.rst
+++ b/changelog/3268.bugfix.rst
@@ -1,0 +1,1 @@
+Fixed test suite to handle "record layer failure" error.

--- a/changelog/3268.bugfix.rst
+++ b/changelog/3268.bugfix.rst
@@ -1,1 +1,1 @@
-Fixed test suite to handle "record layer failure" error.
+Fixed handling of OpenSSL 3.2.0 new error message for misconfiguring an HTTP proxy as HTTPS.

--- a/src/urllib3/connection.py
+++ b/src/urllib3/connection.py
@@ -862,6 +862,7 @@ def _wrap_proxy_error(err: Exception, proxy_scheme: str | None) -> ProxyError:
     is_likely_http_proxy = (
         "wrong version number" in error_normalized
         or "unknown protocol" in error_normalized
+        or "record layer failure" in error_normalized
     )
     http_proxy_warning = (
         ". Your proxy appears to only use HTTP and not HTTPS, "

--- a/test/with_dummyserver/test_socketlevel.py
+++ b/test/with_dummyserver/test_socketlevel.py
@@ -1293,7 +1293,8 @@ class TestSSL(SocketDummyServerTestCase):
         self._start_server(socket_handler)
         with HTTPSConnectionPool(self.host, self.port, ca_certs=DEFAULT_CA) as pool:
             with pytest.raises(
-                SSLError, match=r"(wrong version number|record overflow)"
+                SSLError,
+                match=r"(wrong version number|record overflow|record layer failure)",
             ):
                 pool.request("GET", "/", retries=False)
 

